src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx

```diff
@@ -206,64 +201,61 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
+  )
 
-  const getDefaultOpportunityType = useCallback(
-    (assetId: AssetId) => {
-      const walletSupportsRune = walletSupportsChain({
+  const walletSupportsRune = useMemo(
+    () =>
+      walletSupportsChain({
         chainId: thorchainChainId,
         wallet,
         isSnapInstalled,
-      })
+        chainAccountIds: runeAccountIds,
+      }),
+    [isSnapInstalled, runeAccountIds, wallet],
+  )
 
-      const walletSupportsAsset = walletSupportsChain({
-        chainId: fromAssetId(assetId).chainId,
-        wallet,
-        isSnapInstalled,
-      })
+  const walletSupportsAsset = useMemo(() => {
+    if (!assetId) return false
 
-      const runeSupport = Boolean(
-        walletSupportsRune && accountIdsByAssetId[thorchainAssetId]?.length,
-      )
-      const assetSupport = Boolean(walletSupportsAsset && accountIdsByAssetId[assetId]?.length)
+    const chainId = fromAssetId(assetId).chainId
+    return walletSupportsChain({
+      chainId,
+      wallet,
+      isSnapInstalled,
+      chainAccountIds: poolAssetAccountIds,
+    })
+  }, [assetId, wallet, isSnapInstalled, poolAssetAccountIds])
 
-      if (runeSupport && assetSupport) return 'sym'
-      if (assetSupport) return AsymSide.Asset
-      if (runeSupport) return AsymSide.Rune
-      return 'sym'
-    },
-    [wallet, isSnapInstalled, accountIdsByAssetId],
-  )
+  const getDefaultOpportunityType = useCallback(() => {
+    if (walletSupportsRune && walletSupportsAsset) return 'sym'
+    if (walletSupportsAsset) return AsymSide.Asset
+    if (walletSupportsRune) return AsymSide.Rune
+    return 'sym'
+  }, [walletSupportsRune, walletSupportsAsset])
 
-  // TODO(gomes): Even though that's an edge case for users, and a bad practice, handling sym and asymm positions simultaneously
-  // *is* possible and *is* something that both we and TS do. We can do one better than TS here however:
-  // - When a user deposits symetrically, they can then deposit asymetrically, but only on the asset side
-  // - When a user deposits asymetrically, no matter the side, they *can* deposit symetrically on the other side
-  //   - They can also deposit asymetrically after that, but with one caveat: they can do so only if they deposited asym on the *asset* side only
-  //     In other words, if they have an active asym. RUNE position, they can't deposit symetrically after that unless they withdraw
-  //     The reason for that is that the RUNE side memo performs a nameservice operation, registering the asset address (or a placeholder)
-  //
-  //     We should handle this in the UI and block users from deposits that *will* fail, by detecting their current position(s)
-  //     and not allowing them to select the sure-to-fail deposit types
   useEffect(() => {
+    // No-op if we already have an active opportunity as this effect is just used to set a default
     if (activeOpportunityId) return
+    // No-op if we already have a default i.e we're in a pool view
```

Comment:
It's not quite a no-op if we are actually setting the active opportunity id.

Reply:
Happy to revisit this comment if we have a better wording 🙏🏽 it's a no-op in the sense where we just set the default as the... default, instead of computing one and setting it